docs/topics/auth/default.txt

```diff
@@ -560,8 +595,50 @@ redirects to the login page::
         def my_view(request):
             ...
 
-The permission_required decorator
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. currentmodule:: django.contrib.auth.mixins
+
+.. class:: UserPassesTestMixin
+
+    .. versionadded:: 1.9
+
+    When using :doc:`class-based views </topics/class-based-views/index>`, you
+    can use the ``UserPassesTestMixin`` to do this.
+
+    You have to override the ``test_func`` method of the class to provide
+    the actual test that is performed. Furthermore, you can set any of the
+    parameters of :class:`~django.contrib.auth.mixins.AccessMixin` to customize
+    the handling of unauthorized users::
+
+        from django.contrib.auth.mixins import UserPassesTestMixin
+
+        class MyView(UserPassesTestMixin, View):
+
+            def test_func(self):
+                return self.request.user.email.endswith('@example.com')
+
+    .. note::
```

Comment:
a new habit I'm trying to get into is to use something like `.. admonition: Stacking UserPassesTestMixin` since it gives a more helpful title than "Note"